src/main/java/org/mule/module/xml/internal/util/FileSchemeCorrectionOutputUriResolver.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.xml.internal.util;
+
+import javax.xml.transform.Result;
+
+import net.sf.saxon.lib.StandardOutputResolver;
+import net.sf.saxon.trans.XPathException;
+
+public class FileSchemeCorrectionOutputUriResolver extends StandardOutputResolver {
+
+  private static final String PATH_URI_SCHEME = "file:";
+  private static final String HAS_URI_SCHEME_REGEX = "^[a-zA-Z]+:.+$";
+
+  @Override
+  public Result resolve(String href, String base) throws XPathException {
+    // TODO: Check this!
+    // If neither baseUri is not null, nor the href has an URI scheme, the href will be
+    // considered as an absolute file path
+    if (base == null && !hasUriScheme(href)) {
+      href = PATH_URI_SCHEME.concat(href);
+    }
+    return super.resolve(href, base);
+  }
+
+  protected boolean hasUriScheme(String href) {
+    return href.matches(HAS_URI_SCHEME_REGEX);
```

Comment:
sure this is right? Where did you get this regex from? Seems to me like it can potentially match things that are garbage. Why not just do `new URI(String)` (or similar) and check it doesn't blow up?